components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/GleanLifecycleObserver.kt

```diff
@@ -40,5 +40,14 @@ internal class GleanLifecycleObserver : LifecycleObserver {
         // because it belongs to the baseline ping and that ping is sent every
         // time the app goes to background.
         GleanBaseline.duration.start()
+
+        // We check for the metrics ping schedule here because the app could have been in the
+        // background and resumed in which case Glean would already be initialized but we still need
+        // to perform the check to determine whether or not to collect and schedule the metrics ping.
+        // If this is the first ON_START event since the app was launched, Glean wouldn't be
+        // initialized yet.
+        if (Glean.isInitialized()) {
+            Glean.metricsPingScheduler.startupCheck()
```

Comment:
nit: I think we need to rename `startupCheck` since this is no longer something that's only called at startup. What about just `schedule` or `check`?

Reply:
Sounds good to me!